native/cocos/platform/android/AndroidApp.h

```diff
@@ -0,0 +1,180 @@
+     */
+    uint64_t motionEventsCount;
+
+    /**
+     * Pointer to a read-only array of pointers to AndroidKeyEvent.
+     * Only the first keyEventsCount events are valid.
+     */
+    AndroidKeyEvent keyEvents[NATIVE_APP_GLUE_MAX_NUM_KEY_EVENTS];
+
+    /**
+     * The number of valid "Key" events in `keyEvents`.
+     */
+    uint64_t keyEventsCount;
+};
+
+class NativeWindowCache {
+public:
+    explicit NativeWindowCache(uint32_t windowId) : _windowId{windowId} {
+    }
+
+    ~NativeWindowCache() {
+        setSurface(nullptr);
+    }
+
+    inline void resize(uint32_t width, uint32_t height) { _width = width; _height = height;}
+    void setSurface(jobject surface);
+
+    inline ANativeWindow *getNativeWindow() const { return _nativeWindow; }
+    inline int getWindowId() const { return _windowId; }
+    inline int getWidth() const { return _width; }
+    inline int getHeight() const { return _height; }
+
+    JNIEnv *env{nullptr};
+    ANativeWindow* _pendingWindow{nullptr};
+
+private:
+    uint32_t _width{0};
+    uint32_t _height{0};
+    uint32_t _windowId{0};
+    /** When non-NULL, this is the window surface that the app can draw in. */
+    ANativeWindow *_nativeWindow{nullptr};
+};
+
+class AndroidApp {
+public:
+    /**
+     * An optional pointer.
+     */
+    void* userData;
+
+    /**
+     * The global handle on the process's Java VM.
+     */
+    JavaVM* vm;
+
+    /**
+     * JNI context for the main thread of the app.  Note that this field
+     * can ONLY be used from the main thread of the process; that is, the
+     * thread that calls into the GameActivityCallbacks.
+     */
+    JNIEnv* env;
+
+    /**
+     * The Context object handle.
+     */
+    jobject javaContext;
+
+    /** The current configuration the app is running in. */
+    AConfiguration* config;
+
+    /**
+     * Pointer to the Asset Manager instance for the application.  The
+     * application uses this to access binary assets bundled inside its own .apk
+     * file.
+     */
+    AAssetManager* assetManager;
+
+    /**
+     * Current state of the app.  May be either APP_CMD_START,
+     * APP_CMD_RESUME, APP_CMD_PAUSE, or APP_CMD_STOP.
+     */
+    int appState;
+
+    /**
+     * This is non-zero when the application's GameActivity is being
+     * destroyed and waiting for the app thread to complete.
+     */
+    int destroyRequested;
+
+#define NATIVE_APP_GLUE_MAX_INPUT_BUFFERS 2
+
+    /**
+     * This is used for buffering input from GameActivity. Once ready, the
+     * application thread switches the buffers and processes what was
+     * accumulated.
+     */
+    struct android_input_buffer inputBuffers[NATIVE_APP_GLUE_MAX_INPUT_BUFFERS];
+
+    int currentInputBuffer;
+
+    // Below are "private" implementation of the glue code.
+    /** @cond INTERNAL */
+
+    pthread_mutex_t mutex;
+    pthread_cond_t cond;
+
+    pthread_t thread;
+
+    int running;
+    int destroyed;
+    ARect pendingContentRect;
```

Comment:
Do these variables all need to be public?

Reply:
pendingContentRect can be removed temperary, others need to be declare as global.